tests/lax_control_flow_test.py

```diff
@@ -563,6 +563,51 @@ def cfun(x):
     self.assertEqual(fun(2), cfun(2))
     self.assertEqual(fun(3), cfun(3))
 
+  def testSwitchResidualsMerge(self):
```

Comment:
It'd be nice to have more tests, but in the spirit of landing and iterating (with users!) this at least covers the fundamental case!